### PR TITLE
#184 fix circular injection

### PR DESCRIPTION
**Pull request by junoyoon**
This fix includes the issue on #184 

@songeunwoo  Please run this on our dev-cluster with plugins.

### CODE REVIEW

File: ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/PluginManager.java [hunk 1]
```diff
@@ -58,6 +60,9 @@ public void init() {
 	 * Initialize Plugin Framework.
 	 */
 	public void initPluginFramework() {
+		manager = new NGrinderDefaultPluginManager(config, applicationContext);
+		manager.setExtensionFinder(new NGrinderDefaultExtensionFinder(manager));
+		manager.setSpringExtensionFactory(new SpringExtensionFactory(manager));
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/PluginManager.java]: manager.setSpringExtensionFactory(new NGrinderSpringExtensionFactory(manager, applicationContext));

File: ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/extension/NGrinderSpringExtensionFactory.java [hunk 2]
```diff
@@ -1,51 +1,48 @@
...
+
+	private final PluginManager pluginManager;
+
+	private ApplicationContext applicationContext;
+
+	public NGrinderSpringExtensionFactory(PluginManager pluginManager) {
+		super(pluginManager);
+		this.pluginManager = pluginManager;
+	}
+
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/extension/NGrinderSpringExtensionFactory.java]: public NGrinderSpringExtensionFactory(PluginManager pluginManager, ApplicationContext applicationContext) {
		super(pluginManager);
		this.pluginManager = pluginManager;
		this.applicationContext = applicationContext;
	}

File: ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/extension/NGrinderSpringExtensionFactory.java [hunk 3]
```diff
@@ -1,51 +1,48 @@
...
+	protected void setApplicationContext(ApplicationContext applicationContext) {
+		this.applicationContext = applicationContext;
+	}
+
```
Comment (songeunwoo) [inline ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/extension/NGrinderSpringExtensionFactory.java]: remove.

Comment (junoyoon) [inline ngrinder-controller/src/main/java/org/ngrinder/infra/plugin/extension/NGrinderSpringExtensionFactory.java]: OK

**Pull request merged**